src/users/users.controller.js

```diff
@@ -0,0 +1,123 @@
+import {
+  Controller,
+  Dependencies,
+  UseGuards,
+  Get,
+  Post,
+  Put,
+  Delete,
+  Bind,
+  Body,
+  Param,
+  Query,
+  BadRequestException,
+  NotFoundException,
+} from '@nestjs/common';
+import { AuthGuard } from '@nestjs/passport';
+import { UsersService } from './users.service';
+
+@Controller('users')
+@Dependencies(UsersService)
+export class UsersController {
+  constructor(usersService) {
+    this.usersService = usersService;
+  }
+
+  @UseGuards(AuthGuard('HMAC'))
+  @Post('')
+  @Bind(Body())
+  async create(body) {
+    const { id, name, dob, address, description } = body;
+
+    let oldUser;
+    oldUser = await this.usersService.getById(id);
+    if (!!oldUser) {
+      throw new BadRequestException('User already exists.');
+    }
+
+    let user;
+    try {
+      user = await this.usersService.create(id, {
+        name,
+        dob,
+        address,
+        description,
+      });
+    } catch (err) {
+      if (err.name === 'ValidationError') {
+        throw new BadRequestException(err.message);
+      } else {
+        throw err;
+      }
+    }
+    return user.toJSON();
```

Comment:
do you know a mongoose method: lean(), what is the difference between lean(), instance.toJSON() and the plugin you are using for hidden some fields.

Reply:
I'm not familiar with `lean`. 
It seems that `lean` requires less CPU and memory, and will return pure data object including fields like `_id` and `__v`. 
The plugin I use to hide `_id` and `__v` only works with `toJSON` and `toObject`.
When we use `instance toJSON`, we will remove object methods including `instance.update` and `instance.remove`, and get pure data.

When saving machines is really important, I may try `lean` when it is a major bottleneck.
Currently I prefer not to use `lean`, to keep the concepts in this system fewer so that developers can work with it with less mental burden.